rero_mef/concepts/mef/api.py

```diff
@@ -72,13 +72,37 @@ def replace_refs(self):
         """Replace $ref with real data."""
         data = deepcopy(self)
         data = super().replace_refs()
-        sources = []
+        data['sources'] = [
+            concept for concept in self.entities if data.get(concept)]
+        data['type'] = 'bf:Concept'
+        return data
+
+    def add_information(self, resolve=False, sources=False):
+        """Add information to record."""
+        replace_refs_data = deepcopy(self).replace_refs()
+        data = replace_refs_data if resolve else deepcopy(self)
+        my_sources = []
         for concept in self.entities:
-            if data.get(concept):
-                sources.append(concept)
-                if metadata := data[concept].get('metadata'):
-                    data[concept] = metadata
-        data['sources'] = sources
+            if concept_data := data.get(concept):
+                # we got a error status in data
+                if concept_data.get('status'):
+                    data.pop(concept)
+                    current_app.logger.error(
+                        f'MEF replace refs {data.get("pid")} {concept}'
+                        f' status: {concept_data.get("status")}'
+                        f' {concept_data.get("message")}')
+                else:
+                    my_sources.append(concept)
+                for concept in self.entities:
+                    if concept_data := replace_refs_data.get(concept):
+                        if metadata := concept_data.get('metadata'):
+                            data[concept] = metadata
+                        if not data[concept].get('type'):
+                            data[concept]['type'] = 'bf:Concept'
+        if my_sources and (resolve or sources):
+            data['sources'] = my_sources
+        if not data.get('type'):
+            data['type'] = 'bf:Concept'
```

Comment:
Is that "bf:Concept" don't be a constant for all record in this package ?
If yes, you could remove all tests `if x.get('type'): ...` and use direct assignement